### DISCUSSION TITLE
Breeze Theme with Cloudflare + Rocket Loader Not working

### DISCUSSION THREAD

**Discussion by niravGV** [breezefront/community#116] [1. Help] [answered]
Hello @vovayatsyuk 

We are implementing Mobile theme with a Breeze and Desktop theme with Default Magento due to custom requirements.

We have Cloudflare and along with **Rocket Loader Improve the paint time for pages which include JavaScript.**

I am attaching a screenshot of the showing error in console as per the error i am debugging the the values are getting undefined when this js file calling.

it mostly happen with the **window.checkout** and **window.sectionsConfig** are not getting when the js calling.

Can you please me if you or anyone face such kind of challenges?

![Screenshot from 2023-05-02 12-25-21](https://user-images.githubusercontent.com/30285024/235601986-83cfa650-2b22-468f-a3a9-bed7f0bfbc16.png)
![Screenshot from 2023-05-02 12-24-57](https://user-images.githubusercontent.com/30285024/235602000-fbfa5d40-0ce9-442c-a411-af04f6b4e574.png)

Thanks
Nirav.

**Comment by vovayatsyuk** [comment]
Do not use rocket loader, or any other third-party js deferring tools. Have a look at our performance optimization page: https://breezefront.com/docs/performance

**Comment by niravGV** [comment]
Hello @vovayatsyuk 

Problem is that we are using for Desktop theme since longer Rocket loader, client denied to stop Rocket loader.
But we need to make it working with a Breeze theme with Enable Rocket loader.

Please help me in this, is there any case or any solution for this?

Thanks

**Reply by vovayatsyuk** [reply]
Sorry, I don't have any experience with rocket loader and don't have a demo store to see if it's possible to make it work with breeze. Can you share your URL?

**Comment by niravGV** [comment]
Hello @vovayatsyuk 

I have shared URL in your twitter account personally, if you can check and and little debug can help.

Thanks
Nirav.

**Reply by vovayatsyuk** [reply]
For some reason, Breeze scripts are excluded from "rocket loader":

```
<!-- This one is fine. Type is changed -->
<script type="5cabcc35f4f606933e493bdb-text/javascript" src="...js/firebase-app.min.js"></script>

<!-- this one is changed too -->
<script data-breeze type="5cabcc35f4f606933e493bdb-text/javascript"> var customerDataConfig = { ... }</script>

<!-- This one is skipped. -->
<script data-breeze defer data-cfasync="false" src="...Swissup_Breeze/js/components/customer-data.min.js"></script>
```

Do you know how the rocket loader works and why it doesn't process all breeze scripts?

**Comment by niravGV** [comment]
Hello @vovayatsyuk 

We have done that because of none of script working when adding rocket loader so we have added flag data-cfasync="false" in app/code/Swissup/Breeze/Block/Js.php file from where script are loading.

Do you think we need to add that in rest of script as well ?

Here We have two options 

1. I will remove that flag from everywhere where i have set in the JS files
2. I will add that flag to the customerDataConfig script tag

Let me try first second one as its only few errors if that don't work then i will try first one and let you know update here.

One more curious question 

`<script data-breeze type="5cabcc35f4f606933e493bdb-text/javascript"> var customerDataConfig = { ... }</script>` how you observed this as i can't see this changes visible at my end.

Thanks.
Nirav

**Reply by vovayatsyuk** [reply, marked as answer]
To achieve option 2, need to add the same `data-cfasync="false"` attribute for all inline scripts:

- https://github.com/breezefront/module-breeze/blob/master/view/frontend/templates/theme/js/require_js.phtml#L23-L27
- https://github.com/breezefront/module-breeze/blob/master/view/frontend/templates/theme/js/cookie.phtml#L5
- https://github.com/breezefront/module-breeze/blob/master/view/frontend/templates/customer/js/customer-data.phtml
- https://github.com/breezefront/module-breeze/blob/master/view/frontend/templates/customer/js/section-config.phtml#L14
- https://github.com/breezefront/theme-frontend-breeze-blank/blob/master/Magento_Checkout/templates/cart/minicart.phtml#L49-L56

**Comment by niravGV** [comment]
Thanks for your help @vovayatsyuk  i am checking that with the modification also i am doing with my custom js changes as well once it will apply and upload on the test server i will inform you to check if facing any error, hope this solution works..

**Comment by niravGV** [comment]
Hello @vovayatsyuk 

This second solution works well and working fine now.
You can also check on your end as i have given you end point of test store.

Thank you for your quick help and response.. really appreciate your efforts.
